### ISSUE TITLE
RegEx with indexes is broken

### ISSUE THREAD

**Issue by ZNicoZ** [sakuli/sakuli#97]
**Version**
2.0.0
**Short overview**
RegEx is working differently and is not working at all with indexes
**Detailed error description**
RegEx expressions syntax in Sakuli 1 was "/something/", now in Sakuli 2 it seems to be /something/.
This makes tests using RegEx not directly compatible to Sakuli 2.
Furthermore, the use of indexes won´t work with this new syntax.
**Steps to reproduce error**
```
(async () => { 
    const testCase = new TestCase(); 
    const url = "https://sakuli.io";
    try {
        await _navigateTo(url + "/enterprise");
        await _highlight(_span(/aintenance/)); // working but not directly compatible to Sakuli 1
        await _highlight(_span("/aintenance/")); // not working
        await _highlight(_span("/aintenance/[0]")); // not working
        await _highlight(_span(/aintenance/[0])); // not working
        await _highlight(_span(/aintenance[0]/)); // not working

    } catch (e) {
        await testCase.handleException(e); 
    } finally {
        await testCase.saveResult(); 
    }
})().then(done); 
```

**Comment by tnobody** [comment]
Thank you for this issue.

TLDR;

```javascript
await _highlight(_span("/aintenance/")); // not working
await _highlight(_span("/aintenance/[0]")); // not working
```

This cases seems not to work as expected and we will work on it.

---

Let me explain your examples. Please also see [sahi docs](https://sahipro.com/docs/sahi-apis/accessor-api-basics.html#Identifying%20unique%20element). There are some good examples (unfortunately there is no formal specification for the behavior) 

---

```typescript
await _highlight(_span(/aintenance/)); // working but not directly compatible to Sakuli 1
```

This is a valid way to fetch an element in Sahi (Sakuli 1). The syntax is a [native feature of JavaScript](http://speakingjs.com/es5/ch01.html#basic_regexps)

---

```typescript
await _highlight(_span("/aintenance/")); // not working
````
 This should basically work. We will analyse this.

---

```typescript
await _highlight(_span("/aintenance/[0]")); // not working
```

This is not correctly implemented so far

---

```typescript
await _highlight(_span(/aintenance/[0])); // not working
```

Since the syntax `/.../` creates an RegExp  object (similar to `new RegExp(...)`) you are trying to accessing a property `0` of this RegExp instance. This value resolves to `undefined` so this expression is effectively:

```typescript
await _highlight(_span(undefined)); // not working
```

---

```typescript
await _highlight(_span(/aintenance[0]/)); // not working
```

In this case `[0]`is part of the RegEpx itself and therefore part of the validation. [Brackets are part of RegEpx syntax](https://www.regular-expressions.info/charclass.html) and define so called Charater classes (e.g. digits from 0-9 is [0-9]). The specified character group just matches a `0` in this case.

```typescript
/aintenance[0]/.test('aintenance0') // -> true
```
